src/settings/BatchGroupConfigurationSettings/utils.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+  keys,
+  pickBy,
+} from 'lodash';
+
+export const isUploadURIValid = (value) => {
+  const REGEXP_URI = new RegExp('^$|([f][t][p])([s])?://.+$');
+  const isTrue = REGEXP_URI.test(value);
+
+  if (value === undefined || isTrue) return undefined;
+
+  return <FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.isUploadURIValid" />;
+};
+
+export const saveExportConfig = (formValues, mutator, selectedBatchGroupId, fetchExportConfig, showCallout) => {
+  const credentialsMethod = formValues.credentialsId ? 'PUT' : 'POST';
+
+  const config = {
+    id: formValues.id || null,
+    batchGroupId: selectedBatchGroupId,
+    startTime: formValues.startTime || null,
+    weekdays: keys(pickBy(formValues.weekdays)),
+    enableScheduledExport: formValues.enableScheduledExport,
+    format: formValues.format,
+    uploadURI: formValues.uploadURI,
+  };
+
+  const credentials = {
+    username: formValues.username,
+    password: formValues.password,
+    exportConfigId: formValues.id,
+  };
+
+  if (formValues.id) {
+    mutator.exportConfig.PUT(config)
+      .then(() => {
+        if (formValues.username) {
+          return mutator.credentials[credentialsMethod](credentials);
+        } else {
+          return Promise.resolve();
+        }
+      })
+      .then(() => {
+        fetchExportConfig(selectedBatchGroupId);
+        showCallout({
+          messageId: 'ui-invoice.settings.batchGroupConfiguration.save.success',
+          type: 'success',
+        });
+      })
+      .catch(() => showCallout({
+        messageId: 'ui-invoice.settings.batchGroupConfiguration.save.error',
+        type: 'error',
+      }));
+  } else {
+    mutator.exportConfig.POST(config)
+      .then(({ id }) => {
+        mutator.exportConfigId.update({ id });
+
+        return id;
+      })
+      .then(id => {
+        if (formValues.username) {
```

Comment:
what if there is no username, just password?

Reply:
I think you can look at https://github.com/folio-org/ui-organizations/blob/master/src/interfaces/EditInterface/util.js#L3 as a reference, however I don't expect that it could be just copy-pasted - requirements might be different

Reply:
Thanks for the link, fixed